### PR TITLE
Flask: extend _bulk endpoint to accept json

### LINKED ISSUE #160: Extend /api/_bulk to accept JSON
Content-Type: application/json

Body is an array of key-value pair objects, the keys being exact names of columns among any level of the dataset hierarchy. Logic is the same as with CSV.

Task points: 2

### PR DESCRIPTION
**Pull request by delvinso**
```
# just json 
curl -X POST \
    -H "Content-Type: application/json"  \
    --data '{
                "family_codename": "1001",
                "participant_codename":"06332",
                "tissue_sample" : "Blood",
                "tissue_sample_type" :"Blood",
                "dataset_type" : "WGS",
                "condition": "GermLine",
                "gender": "Male",
                "participant_type": "Parent"
            }' \
    localhost:5000/api/_bulk

JSON must be in an array%   
```
```
# array of json (1)
curl -X POST \
    -H "Content-Type: application/json"  \
    --data '[{
                "family_codename": "1001",
                "participant_codename":"06332",
                "tissue_sample" : "Blood",
                "tissue_sample_type" :"Blood",
                "dataset_type" : "WGS",
                "condition": "GermLine",
                "gender": "Male",
                "participant_type": "Parent"
            }]' \
    localhost:5000/api/_bulk
```
```
# array of json (2)
curl -X POST -H "Content-Type: application/json"  --data '[
                                                            {
                                                                "family_codename": "1001",
                                                                "participant_codename":"1411",
                                                                "tissue_sample" : "Blood",
                                                                "tissue_sample_type" :"Blood",
                                                                "dataset_type" : "WGS",
                                                                "condition": "GermLine"
                                                            },
                                                            {
                                                                "family_codename": "1001",
                                                                "participant_codename":"3420",
                                                                "tissue_sample" : "Blood",
                                                                "tissue_sample_type" :"Blood",
                                                                "dataset_type" : "WES",
                                                                "condition": "GermLine"
                                                            }
                                                        ]' \localhost:5000/api/_bulk

```

### CODE REVIEW

**Review by kevinlul — CHANGES_REQUESTED** [review]
This seems like it'll only accept a single object rather than an array in the request body?

**Comment by delvinso** [comment]
The endpoint will now accept single or multiple json objects as long as they are in an array.

File: flask/app/routes.py [hunk 1]
```diff
@@ -311,24 +311,35 @@ def bulk_update():
             "notes",
         ],
     }
+    if request.content_type not in ["text/csv", "application/json"]:
+        return "Only Content Type 'text/csv' or 'application/json' Supported", 415
```
Comment (kevinlul) [inline flask/app/routes.py:315]: This could be done as the final `else:` clause for the if/elif following

**Review by kevinlul — APPROVED** [review]
Looks great! This was simpler than I thought it would be.

File: flask/app/routes.py (hunk 1 above)
Comment (delvinso) [reply flask/app/routes.py:315]: good point, I actually thought the same since it seemed out of place at the beginning

**Pull request merged**